### PR TITLE
enable custom links for more component model subclasses

### LINKED ISSUE #2946: Add custom link feature to dcim.interfaces
### As ...

Nelly - Network Engineer

### I want ...

... to be able to add custom links to dcim.interfaces (on the specific interface page and not the "all interfaces" overview page)

### So that ...

... I'm able to add a link to an external application (e.g. external IPAM) which allows the user to click to get additional information about the end device that is connected to the interface x of device y

### I know this is done when...

... I'm able to add custom links for dcim.interfaces content types (not the overview page but the individual interface page)

### Optional - Feature groups this request pertains to.

- [ ] Automation
- [ ] Circuits
- [X] DCIM
- [ ] IPAM
- [ ] Misc (including Data Sources)
- [ ] Organization
- [ ] Plugins (and other Extensibility)
- [ ] Security (Secrets, etc)
- [ ] Image Management
- [ ] UI/UX
- [ ] Documentation
- [ ] Other (not directly a platform feature)

### Database Changes

This feature already exists for a some of the content types like devices, cables, locations, ipam, etc.

I don't know what changes must be made to extend this feature for dcim.interfaces

### External Dependencies

_No response_

### PR DESCRIPTION
**Pull request by teunvink**
# Closes #2946
# What's Changed

Add  the `custom_links` to more ComponentModel subclasses, so we can create custom links for e.g. device interfaces. 

# Screenshots

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example App Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/dcim/models/device_components.py [hunk 1]
```diff
@@ -265,6 +266,8 @@ def parent(self):
 
 @extras_features(
     "cable_terminations",
+    "custom_fields",
```
Comment (glennmatthews) [inline nautobot/dcim/models/device_components.py:269]: Shouldn't need to add this - `custom_fields` is implicit on all models that support them for the past few releases.

**Review by glennmatthews — COMMENTED** [review]
Looks great!

- Can you please add a `changes/2946.added` [changelog fragment](https://docs.nautobot.com/projects/core/en/stable/development/core/#creating-changelog-fragments) to the PR?
- Is there a reason you didn't add custom links to `ConsoleServerPort` as well?

Thanks!

File: nautobot/dcim/models/device_components.py (hunk 1 above)
Comment (teunvink) [reply nautobot/dcim/models/device_components.py]: removed it.

**Comment by teunvink** [comment]
>     * Is there a reason you didn't add custom links to `ConsoleServerPort` as well?

There isn't, I missed it and added it now.

File: changes/2946.added [hunk 2]
```diff
@@ -0,0 +1 @@
+Allow custom links for interfaces, console ports, power ports, power outlets, front ports, rear ports and device bays.
```
Comment (glennmatthews) [inline changes/2946.added:1]:
```suggestion
Added custom link support for interfaces, console ports, console server ports, power ports, power outlets, front ports, rear ports, device bays, and inventory items.
```

**Review by glennmatthews — APPROVED** [review]

**Review by gsnider2195 — APPROVED** [review]

**Pull request merged**